### DISCUSSION TITLE
Question about onRecordAfterCreateRequest hook logic isolation

### DISCUSSION THREAD

**Discussion by imfurman** [pocketbase/pocketbase#5735] [Q&A] [answered]
I’m using a onRecordAfterCreateRequest hook to send additional data after creating a record. The question is whether these hooks are isolated — for instance, if two users make requests simultaneously, will each work only with their own record, or is there a risk of them receiving each other’s data, meaning validation is required?

**Comment by ganigeorgiev** [comment]
It is not clear what actually is your concern and what do you mean by "_risk of them receiving each other’s data, meaning validation is required_". What validations do you have in mind?

Each record request hook has its own unique event argument. If the server receives 2 create requests and they complete successfully then `onRecordAfterCreateRequest` should fire 2 times, each with different `e.record`.

**Reply by imfurman** [reply]
Apologies if my writing isn’t clear; English isn’t my native language. In short, I see that the hook depends on the entire collection, so I thought that a request from one client might affect a request from another. To put it differently: let’s say the first client’s request takes 15 seconds, and in the meantime, a second client’s request comes in—will the first request affect the second, or will the second request be independent and finish based on its own response time?

**Reply by ganigeorgiev** [reply, marked as answer]
> will the first request affect the second

No, all requests and their events are processed separately. If one request context somehow hold a stale data from another then that would be a bug (and a pretty serious one).

> will the second request be independent and finish based on its own response time?

I'm still not entirely sure what do you mean by "be independent". While requests execute concurrently there is no guarantee that they will be processed in parallel, meaning that it is possible one request to block the execution of another.

The Go std `net/http` by default processes each request in a separate goroutine concurrently but again whether this will happen in parallel/simultaneously depends on the Go runtime and other factors that we don't directly control (the number of available threads, configuration options like `GOMAXPROCS`, etc.)

**Reply by imfurman** [reply]
Thank you for the clarification and for the excellent product!
